src/r5py/r5/transport_network.py

```diff
@@ -110,6 +122,44 @@ def __exit__(self, exception_type, exception_value, traceback):
         """Exit context."""
         return False
 
+    @functools.cached_property
+    def _cache_directory(self):
+        cache_dir = (
+            pathlib.Path(config.CACHE_DIR)
+            / f"{self.__class__.__name__:s}_{hash(self):x}"
+        )
+        cache_dir.mkdir(exist_ok=True)
+        return cache_dir
+
+    def _working_copy(self, input_file):
+        """Create a copy or link of an input file in a cache directory.
+
+        This method exists because R5 creates temporary files in the
+        directory of input files. This can not only be annoying clutter,
+        but also create problems of concurrency, performance, etc., for
+        instance, when the data comes from a shared network drive or a
+        read-only file system.
+
+        Arguments
+        ---------
+        input_file : str or pathlib.Path
+            The file to create a copy or link of in a cache directory
+
+        Returns
+        -------
+        pathlib.Path
+            The path to the copy or link created
+        """
+        # try to first create a symbolic link, if that fails (e.g., on Windows),
+        # copy the file to a cache directory
+        input_file = pathlib.Path(input_file)
+        destination_file = pathlib.Path(self._cache_directory / input_file.name)
+        try:
```

Comment:
Hmm, I'm not a big fan of implementing logic using try and catch. Could we write this in a way that we detect whether the operating system is Windows and then do the copy/paste and otherwise create a symbolic link?

Reply:
That’s now a philosophical one, and I disagree with your stance here: Symbolic links are the way to go, except if an error occurs because operating system conditions are not met. To catch this anticipated error using an exception and in the exception attempting the (more fail-safe) second-best option is exactly the correct pattern, IMO.

This is at the core of the Pythonic philosophy of ‘Ask forgiveness, not permission’. Explicit logic for testing OS capabilities is not.

Reply:
Sorry to continue on this, but this is definitely the OPPOSITE of risky. This is trying the best-suited approach, and having a fallback if it is not possible.

Reply:
Okay, fair enough 🙂 👍🏻